core/arch/arm/plat-rockchip/platform_rk3566.c

```diff
@@ -0,0 +1,44 @@
+#include <common.h>
+#include <io.h>
+#include <kernel/panic.h>
+#include <mm/core_memprot.h>
+#include <platform.h>
+#include <platform_config.h>
+
+#define SGRF_DDRRGN_CON0_16(n)		((n) * 4)
```

Comment:
And this one I do not understand. This seems to be some RK3399 registers writing, but I have not been able do understand what exactly is going on here so far.

Reply:
This seems to be a helper macro for DDR region registers as stated [here](https://gist.github.com/CrystalGamma/a68333fa4c9fda7eb6c09d30ad4937fe). And [there is](https://github.com/rockchip-linux/u-boot/blob/63c55618fbdc36333db4cf12f7d6a28f0a178017/arch/arm/include/asm/arch-rockchip/grf_rk3399.h#L274) a precise address of the SGRF registers for rk3399, but I have not been able to find these for rk3566 or rk3568 so far :cry: .